### PR TITLE
feat: get cm from apiserver cache directly to reduce apiserver and etcd load

### LINKED ISSUE #120126: kubelet get configmap from kube-apiserver and etcd everytime
### What happened?

qps of configmap get request is too high in kube-apiserver.

when there are many configmaps in cluster mounted by running pod, the qps of configmap get request will be high and the request will also need to get configmap resource from etcd instead of returned from kube-apiserver cache.

### What did you expect to happen?

DO NOT get configmap from etcd. Get it from cache of kube-apiserver.

### How can we reproduce it (as minimally and precisely as possible)?

1. set kubelet configMapAndSecretChangeDetectionStrategy to Cache;
2. set kube-apiserver --v=3
3. create lots of configmaps mount to pods
4. check kube-apiserver metrics `sum(rate(apiserver_request_total{resource="configmaps",verb="GET"}[5m])) by (verb,resource,subresource)` 
5. check kube-apiserver log

### Anything else we need to know?

I have tested it in v1.22.13 and v1.28.0. The logic to set resourceversion=0 seems never be triggered

### Kubernetes version

<details>

```console
1.22.13, 1.28.0 
```

</details>

### Cloud provider

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by likakuli**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
change cachebasedmanager to get configmap from apiserver cache directly always to reduce apiserver and etcd load.
#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/120126

#### Special notes for your reviewer:
Currently, cachebasedmanager has a logic to get configmap from apiserver cache but it will not be triggered. `sum(rate(apiserver_request_total{resource="configmaps",verb="GET"}[5m])) by (verb,resource,subresource)` is high and there will be a lot of etcd request for configmap resource same time.
#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/util/manager/cache_based_manager.go [hunk 1]
```diff
@@ -178,15 +177,10 @@ func (s *objectStore) Get(namespace, name string) (runtime.Object, error) {
 	data.Lock()
 	defer data.Unlock()
 	if data.err != nil || !s.isObjectFresh(data) {
-		opts := metav1.GetOptions{}
-		if data.object != nil && data.err == nil {
-			// This is just a periodic refresh of an object we successfully fetched previously.
-			// In this case, server data from apiserver cache to reduce the load on both
-			// etcd and apiserver (the cache is eventually consistent).
-			util.FromApiserverCache(&opts)
```
Comment (matthyx) [inline pkg/kubelet/util/manager/cache_based_manager.go:186]: @likakuli are you sure this is causing load on apiserver?

Comment (likakuli) [reply pkg/kubelet/util/manager/cache_based_manager.go:186]: yeah, can be reproduced easy in master branch. When there are many configmaps in cluster which mounted to running pods, there will be too many api calls to kube-apiserver to get configmap like below
```
2023-08-23T08:52:56.426912273Z stderr F I0823 08:52:56.426704       1 httplog.go:132] "HTTP" verb="GET" URI="/api/v1/namespaces/default/configmaps/nginx-cfgmap" latency="1.889343ms" userAgent="kubelet/v1.28.0 (linux/amd64) kubernetes/855e7c4" audit-ID="d42c3ca8-a6c5-482d-a9a1-7377f35f8a83" srcIP="192.168.228.2:59052" apf_pl="system" apf_fs="system-nodes" apf_iseats=1 apf_fseats=0 apf_additionalLatency="0s" apf_execution_time="1.287162ms" resp=200
2023-08-23T08:54:26.324041175Z stderr F I0823 08:54:26.323627       1 httplog.go:132] "HTTP" verb="GET" URI="/api/v1/namespaces/default/configmaps/nginx-cfgmap" latency="2.207789ms" userAgent="kubelet/v1.28.0 (linux/amd64) kubernetes/855e7c4" audit-ID="36698661-5b71-4467-a107-77ee97096788" srcIP="192.168.228.2:59052" apf_pl="system" apf_fs="system-nodes" apf_iseats=1 apf_fseats=0 apf_additionalLatency="0s" apf_execution_time="1.533643ms" resp=200
2023-08-23T08:55:54.331196195Z stderr F I0823 08:55:54.330840       1 httplog.go:132] "HTTP" verb="GET" URI="/api/v1/namespaces/default/configmaps/nginx-cfgmap" latency="1.926865ms" userAgent="kubelet/v1.28.0 (linux/amd64) kubernetes/855e7c4" audit-ID="36cfcbe3-d76a-4a4d-b251-47cc2df060cb" srcIP="192.168.228.2:59052" apf_pl="system" apf_fs="system-nodes" apf_iseats=1 apf_fseats=0 apf_additionalLatency="0s" apf_execution_time="1.269527ms" resp=200
2023-08-23T08:57:09.333913507Z stderr F I0823 08:57:09.333470       1 httplog.go:132] "HTTP" verb="GET" URI="/api/v1/namespaces/default/configmaps/nginx-cfgmap" latency="1.810334ms" userAgent="kubelet/v1.28.0 (linux/amd64) kubernetes/855e7c4" audit-ID="563bd337-df29-4342-afd0-9ca6e0632f0f" srcIP="192.168.228.2:59052" apf_pl="system" apf_fs="system-nodes" apf_iseats=1 apf_fseats=0 apf_additionalLatency="0s" apf_execution_time="1.177012ms" resp=200
```

**Comment by matthyx** [comment]
I don't think this PR solves the problem you're describing in the comment...

I suggest you open an issue instead and put these logs along with detailed information for reproduction.

**Comment by likakuli** [comment]
ok opened https://github.com/kubernetes/kubernetes/issues/120126

**Comment by matthyx** [comment]
Thanks, let's close this one and wait for issue triage.
/close

**Pull request closed without merging**